homeassistant/components/climate/mqtt.py

```diff
@@ -116,6 +119,10 @@
     vol.Optional(CONF_SEND_IF_OFF, default=True): cv.boolean,
     vol.Optional(CONF_PAYLOAD_ON, default="ON"): cv.string,
     vol.Optional(CONF_PAYLOAD_OFF, default="OFF"): cv.string,
+
+    vol.Optional(CONF_MIN_TEMP): vol.Coerce(float),
```

Comment:
So I think an even cleaner solution would be to write

```python
vol.Optional(CONF_MIN_TEMP, default=DEFAULT_MIN_TEMP): vol.Coerce(float),
```

The later, you could just write:

```python
@property
def min_temp(self):
    """Return the minimum temperature."""
    return self._min_temp
```

The voluptuous schema is where usually all default values should reside in (when there are config schemas). Having the default values in code usually leads to bugs one way or another.

Reply:
Banged my head against this too.

DEFAULT_*_TEMP is in CELCIUS, CONF_MIN_TEMP might not be... using the default=DEFAULT_*_TEMP leaves an innuendo on what is the unit of the temperature voluptuous returns :(

Reply:
Then again if I revert it to `super().min_temp` it will do the conversion according to `convert_temperature()` and return that value. Or implement that logic in each subclass?